src/main/scala/com/microsoft/hyperspace/index/IndexCollectionManager.scala

```diff
@@ -63,13 +64,12 @@ class IndexCollectionManager(
     }
   }
 
-  override def refresh(indexName: String): Unit = {
+  override def refresh(indexName: String, mode: String = REFRESH_MODE_INCREMENTAL): Unit = {
     withLogManager(indexName) { logManager =>
       val indexPath = PathResolver(spark.sessionState.conf).getIndexPath(indexName)
       val dataManager = indexDataManagerFactory.create(indexPath)
-      if (HyperspaceConf.refreshDeleteEnabled(spark)) {
+      if (mode.equals(REFRESH_MODE_INCREMENTAL)) {
         new RefreshDeleteAction(spark, logManager, dataManager).run()
```

Comment:
I think we barely need to merge these two actions as `RefreshAppendAction` only handles newly appended files - no duplicate index data write.

+ BTW it might be good to log the number(and size) of appended files & deleted files in info level?

Reply:
> I think we barely need to merge these two actions as `RefreshAppendAction` only handles newly appended files - no duplicate index data write.

sorry i didn't understand what you mean by this. Are you saying what I am doing is correct?

Reply:
Yes, means this approach looks fine. I just wanted to correct myself :)

Reply:
Sure thanks for confirming! I will add the log lines as well. thanks!